DriverDataUI/src/main/java/com/drivequant/drivekit/ui/mysynthesis/MySynthesisViewModel.kt

```diff
@@ -86,13 +98,23 @@ internal class MySynthesisViewModel(application: Application) : AndroidViewModel
                 this.selectedDate = date
                 this.dateSelectorViewModel.configure(dates, selectedDateIndex, this.selectedPeriod)
 
+                val currentAllContextItemIndex = timelineSource.allContext.indexOfFirst { it.date == this.selectedDate }
+
                 this.scoreCardViewModel.configure(
-                    this.selectedScore,
-                    this.selectedPeriod,
-                    timelineSource.getDriverScoreSynthesis(this.selectedScore, date),
-                    timelineSource.allContext.first { it.date == this.selectedDate }
+                    score = this.selectedScore,
+                    period = this.selectedPeriod,
+                    scoreSynthesis = timelineSource.getDriverScoreSynthesis(this.selectedScore, date),
+                    allContextItem = timelineSource.allContext[currentAllContextItemIndex],
+                    previousDate = if (currentAllContextItemIndex >= 1) { timelineSource.allContext[currentAllContextItemIndex - 1].date } else null
```

Comment:
Maintenant tu peux utiliser les extensions des modules internes :
- `timelineSource.allContextItemAt(date)` (pour `allContextItem`)
- `timelineSource.previousAllContextItemFrom(date)?.date` (pour `previousDate`)

Ce qui permet de simplifier et de supprimer `val currentAllContextItemIndex = timelineSource.allContext.indexOfFirst { it.date == this.selectedDate }`.

Reply:
C'est mis à jour, merci